### ISSUE TITLE
LiquidTags.extendMarkdown() missing 1 required positional argument: 'md_globals'

### ISSUE THREAD

**Issue by maphew** [pelican-plugins/liquid-tags#38]
I'm getting this error on a site created with _[blog-o-matic](https://github.com/iranzo/blog-o-matic)_ template.

```
(pelican) C:\Users\Matt\web\mhw.com-elegant>pelican
[10:01:44] WARNING  Removed extraneous trailing slash from SITEURL.                                                                                                                         log.py:91
           WARNING  Feeds generated without SITEURL set properly may not be valid                                                                                                           log.py:91
           ERROR    Cannot register plugin `tipue_search`                                                                                                                                   log.py:96
                    module 'tipue_search' has no attribute 'register'
[10:01:46] ERROR    Could not process Linux\change-dir-name-case.md                                                                                                                         log.py:96
                    LiquidTags.extendMarkdown() missing 1 required positional argument: 'md_globals'
           ERROR    Could not process Projects\Canvec\Using_CanVec.md                                                                                                                       log.py:96
                    LiquidTags.extendMarkdown() missing 1 required positional argument: 'md_globals'
```
...repeats for each document in `content`.

Versions:
- Python 3.10.8
- markdown 3.4.1

I don't know how to get the installed liquid_tags version, so can't tell if the fix in #20 is present.

**Comment by maphew** [comment]
I think I've got an older version of liquid tags. If I'm interpreting https://github.com/pelican-plugins/liquid-tags/pull/20/files#diff-64678f0e067a6e5b345f3dd61ee483febe3abcca283e00415117217081733ab3R89 lines 85/89 correctly what I have in my tree is old:

```
    def extendMarkdown(self, md, md_globals):
...snip...
```

So perhaps I've created an issue in the wrong project. If so, which is the right one? Pelican-plugins? Blog-o-matic? Elegant? I'm not sure how these relate to each other. I see there's `peru` for dependency management, which is new to me. Maybe I need to be leveraging it somehow. Anyway, if this is the wrong place, sorry for the noise, and please point me in the right direction. Thanks!

**Comment by mirekdlugosz** [comment]
Yes, this is issue #19 that we have fixed in #20. Your setup is currently using new Pelican, new Markdown and old liquid-tags. The easiest way of getting out of this is pulling in old Markdown version:

```
pip install markdown==3.3.7
```

You can modify your `requirements.txt` to pin Markdown.

---------------

Of course the real question is why you end up with old liquid-tags. Apparently, [blog-o-matic is using pelican-plugins repository](https://github.com/iranzo/blog-o-matic/blob/master/peru.yaml#L30), which has [2-year old liquid-tags](https://github.com/getpelican/pelican-plugins/tree/master/liquid_tags). Pelican community started to move away from this repository into separate pelican-plugins organization with [Pelican 4.5, released in 2020](https://getpelican.com/blog/pelican-4.5-released/).

For me, this is blog-o-matic issue. Their configuration results in usage of outdated packages. They either need to pin Pelican and Markdown to pre-2020 versions, or they need to start using plugins from new repositories.

**Comment by maphew** [comment]
Ahh, thank you for the history and status quo change! I had completely missed the shift away from monorepo. I will redirect to blog-o-matic and related.
